daemon/src/db.rs

```diff
@@ -1,61 +1,63 @@
 
 pub async fn run_migrations(pool: &SqlitePool) -> anyhow::Result<()> {
     sqlx::migrate!("./migrations").run(pool).await?;
     Ok(())
 }
 
 pub async fn insert_order(order: &Order, conn: &mut PoolConnection<Sqlite>) -> anyhow::Result<()> {
-    let uuid = serde_json::to_string(&order.id).unwrap();
-    let trading_pair = serde_json::to_string(&order.trading_pair).unwrap();
-    let position = serde_json::to_string(&order.position).unwrap();
-    let initial_price = serde_json::to_string(&order.price).unwrap();
-    let min_quantity = serde_json::to_string(&order.min_quantity).unwrap();
-    let max_quantity = serde_json::to_string(&order.max_quantity).unwrap();
-    let leverage = order.leverage.0;
-    let liquidation_price = serde_json::to_string(&order.liquidation_price).unwrap();
-    let creation_timestamp = serde_json::to_string(&order.creation_timestamp).unwrap();
-    let term = serde_json::to_string(&order.term).unwrap();
-    let origin = serde_json::to_string(&order.origin).unwrap();
-    let oracle_event_id = order.oracle_event_id.to_string();
-
-    sqlx::query!(
-        r#"
-            insert into orders (
-                uuid,
-                trading_pair,
-                position,
-                initial_price,
-                min_quantity,
-                max_quantity,
-                leverage,
-                liquidation_price,
-                creation_timestamp,
-                term,
-                origin,
-                oracle_event_id
-            ) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
-            "#,
-        uuid,
-        trading_pair,
-        position,
-        initial_price,
-        min_quantity,
-        max_quantity,
-        leverage,
-        liquidation_price,
-        creation_timestamp,
-        term,
-        origin,
-        oracle_event_id
+    sqlx::query(
+        r#"insert into orders (
+            uuid,
+            trading_pair,
+            position,
+            initial_price,
+            min_quantity,
+            max_quantity,
+            leverage,
+            liquidation_price,
+            creation_timestamp_seconds,
+            creation_timestamp_nanoseconds,
+            term_seconds,
+            term_nanoseconds,
+            origin,
+            oracle_event_id
+        ) values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14)"#,
     )
+    .bind(&order.id)
+    .bind(&order.trading_pair)
+    .bind(&order.position)
+    .bind(&order.price.to_string())
+    .bind(&order.min_quantity.to_string())
+    .bind(&order.max_quantity.to_string())
+    .bind(order.leverage.0)
+    .bind(&order.liquidation_price.to_string())
+    .bind(
+        order
+            .creation_timestamp
+            .duration_since(SystemTime::UNIX_EPOCH)?
+            .as_secs() as i64,
+    )
+    .bind(
+        order
+            .creation_timestamp
+            .duration_since(SystemTime::UNIX_EPOCH)?
+            .subsec_nanos() as i32,
```

Comment:
❓ Just out of curiosity: Are nanos actually represented as `i32` or why are we not using `i64` here like for `secs`?

Reply:
`i32`, yes. And not for completely arbitrary reasons either!

```rust
/// A span of time with nanosecond precision.
///
/// Each `Duration` is composed of a whole number of seconds and a fractional part represented in
/// nanoseconds.
///
/// This implementation allows for negative durations, unlike [`core::time::Duration`].
#[derive(Clone, Copy, Default, PartialEq, Eq, Hash, PartialOrd, Ord)]
pub struct Duration {
    /// Number of whole seconds.
    seconds: i64,
    /// Number of nanoseconds within the second. The sign always matches the `seconds` field.
    nanoseconds: i32, // always -10^9 < nanoseconds < 10^9
    #[allow(clippy::missing_docs_in_private_items)]
    padding: Padding,
```